lib/Interpreter/IncrementalParser.cpp

```diff
@@ -125,7 +125,7 @@ namespace {
   if (!II)
     return;
   const clang::DefMacroDirective* MD
-    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
+    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
```

Comment:
It seems the real issue is more subtle. This allows the argument of dyn_cast_or_null to be null. I am assuming that CLING_CXXABIS is not defined in the TU. Perhaps @karies could help.